app/AvailablePlugin/EmailAddressWidget/View/CoEmailAddressWidgets/display.ctp

```diff
@@ -88,7 +94,21 @@
         let xhr = callRegistryAPI(url, 'GET', 'json', this.setEmailAddresses, '', this.generalXhrFailCallback);
       },
       setEmailAddresses: function(xhr){
-        this.emailAddresses = xhr.responseJSON.EmailAddresses;
+        let allEmailAddresses = xhr.responseJSON.EmailAddresses;
+        // limit the email addresses to the emailType:
+        let emailAddressesOfType = [];
+        allEmailAddresses.forEach((address) => {
```

Comment:
use filter instead of forEach, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
Ooh - good point. On it.

Reply:
Done